homeassistant/components/awair/sensor.py

```diff
@@ -1,248 +1,246 @@
+        if self._coordinator.last_update_success and self._air_data:
+            # and the results included our sensor type...
+            if self._kind in self._air_data.sensors:
+                # then we are available.
+                return True
+
+            # or, we're a dust alias
+            if self._kind in DUST_ALIASES and API_DUST in self._air_data.sensors:
+                return True
+
+            # or we are API_SCORE
+            if self._kind == API_SCORE:
+                # then we are available.
+                return True
+
+        # Otherwise, we are not.
+        return False
 
     @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement of this entity."""
-        return self._unit_of_measurement
+    def state(self) -> float:
+        """Return the state, rounding off to reasonable values."""
+        state: float
 
-    async def async_update(self):
-        """Get the latest data."""
-        await self._data.async_update()
+        # Special-case for "SCORE", which we treat as the AQI
+        if self._kind == API_SCORE:
+            state = self._air_data.score
+        elif self._kind in DUST_ALIASES and API_DUST in self._air_data.sensors:
+            state = self._air_data.sensors.dust
+        else:
+            state = self._air_data.sensors[self._kind]
 
+        if self._kind == API_VOC or self._kind == API_SCORE:
+            return round(state)
 
-class AwairData:
-    """Get data from Awair API."""
+        if self._kind == API_TEMP:
+            return round(state, 1)
 
-    def __init__(self, client, uuid, throttle):
-        """Initialize the data object."""
-        self._client = client
-        self._uuid = uuid
-        self.data = {}
-        self.attrs = {}
-        self.async_update = Throttle(throttle)(self._async_update)
+        return round(state, 2)
 
-    async def _async_update(self):
-        """Get the data from Awair API."""
-        resp = await self._client.air_data_latest(self._uuid)
+    @property
+    def icon(self) -> str:
+        """Return the icon."""
+        return SENSOR_TYPES[self._kind][ATTR_ICON]
 
-        if not resp:
-            return
+    @property
+    def device_class(self) -> str:
+        """Return the device_class."""
+        return SENSOR_TYPES[self._kind][ATTR_DEVICE_CLASS]
 
-        timestamp = dt.parse_datetime(resp[0][ATTR_TIMESTAMP])
-        self.attrs[ATTR_LAST_API_UPDATE] = timestamp
-        self.data[ATTR_SCORE] = resp[0][ATTR_SCORE]
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit the value is expressed in."""
+        return SENSOR_TYPES[self._kind][ATTR_UNIT]
 
-        # The air_data_latest call only returns one item, so this should
-        # be safe to only process one entry.
-        for sensor in resp[0][ATTR_SENSORS]:
-            self.data[sensor[ATTR_COMPONENT]] = round(sensor[ATTR_VALUE], 1)
+    @property
+    def attribution(self) -> str:
```

Comment:
This property is never used. We can set the attribution in the `device_state_attributes` instead.

Reply:
Done in 259f3f8f47cb6e3a133ba775ef738804fa25ce44.